src/main/java/uk/gov/pay/directdebit/payments/services/PaymentViewService.java

```diff
@@ -31,26 +40,46 @@ public PaymentViewResponse getPaymentViewResponse(PaymentViewSearchParams search
         return gatewayAccountDao.findByExternalId(searchParams.getGatewayExternalId())
                 .map(gatewayAccount -> {
                     PaymentViewSearchParams validatedSearchParams = paymentViewValidator.validateParams(searchParams);
-                    List<PaymentViewListResponse> viewListResponse = 
-                            getPaymentViewListResponse(paymentViewValidator.validateParams(searchParams));
+                    Long total = getTotal(validatedSearchParams);
+                    if (total == 0) {
+                        throw new RecordsNotFoundException(format("Found no records for %s", validatedSearchParams.buildQueryParamString()));
```

Comment:
I don't understand why we're throwing an exception if we don't find anything? THis is a normal state of affairs, and we should just return and empty array no? (I realise you didn't initiate this pattern)

Reply:
Good point, I shall remove it.